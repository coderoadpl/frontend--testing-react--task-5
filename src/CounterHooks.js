import React from 'react'

import Button from './Button'

export const CounterHooks = () => {
  const [number, setNumber] = React.useState(0)

  const inc = React.useCallback(() => setNumber((prevNumber) => prevNumber + 1), [])

  return (
    <div>
      <h1
        data-testid={'counter__number-display'}
      >
        {number}
      </h1>
      <Button
        data-testid={'counter__button--inc'}
        onClick={inc}
      >
        +
      </Button>
    </div>
  )
}

export default CounterHooks
