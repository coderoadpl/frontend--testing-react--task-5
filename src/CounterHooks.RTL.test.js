import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import CounterHooks from './CounterHooks'

describe('CounterHooks test â€“ React Testing Library', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<CounterHooks />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render valid initial number', () => {
    const { container } = render(<CounterHooks />)

    const numberDisplay = container.querySelector('[data-testid="counter__number-display"]')

    expect(numberDisplay.textContent).toBe('0')
  })

  it('should render valid number after button click', () => {
    const { getByTestId } = render(<CounterHooks />)

    const button = getByTestId('counter__button--inc')

    fireEvent.click(button)

    const numberDisplay = getByTestId('counter__number-display')

    expect(numberDisplay).toHaveTextContent('1')
  })
})
