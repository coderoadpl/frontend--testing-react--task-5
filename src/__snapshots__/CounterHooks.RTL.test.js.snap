// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterHooks test â€“ React Testing Library should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="counter__number-display"
    >
      0
    </h1>
    <button
      data-testid="counter__button--inc"
      style="border-radius: 50%;"
    >
      +
    </button>
  </div>
</DocumentFragment>
`;
